src/ngx_http_lua_ssl_ocsp.c

```diff
@@ -490,7 +490,6 @@ ngx_http_lua_ffi_ssl_set_ocsp_status_resp(ngx_http_request_t *r,
 
     dd("set ocsp resp: resp_len=%d", (int) resp_len);
     (void) SSL_set_tlsext_status_ocsp_resp(ssl_conn, p, resp_len);
-    ssl_conn->tlsext_status_expected = 1;
```

Comment:
Can you please be more specific on how this is needed?

The status_cb is called *after* the cert_cb (meaning that when the status_cb is called the ocsp response is already set), The logic that OpenSSL uses to decide whether to send the response is: if callback returns OK (which in out case is always) and if the OCSP response is not null, then send it. The logic is the same in v1.0.1, v1.0.2 and master:
https://github.com/openssl/openssl/blob/OpenSSL_1_0_1-stable/ssl/t1_lib.c#L1971
https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/ssl/t1_lib.c#L3077
https://github.com/openssl/openssl/blob/master/ssl/t1_lib.c#L2801

Of course I could be wrong, will need to do some more testing I guess (too bad the test suite doesn't support OCSP stapling).